src/go/rpk/pkg/cli/cmd/redpanda/config.go

```diff
@@ -67,6 +67,7 @@ partial json/yaml config objects:
 			p := config.ParamsFromCommand(cmd)
 			cfg, err := p.Load(fs)
 			out.MaybeDie(err, "unable to load config: %v", err)
+			cfg = cfg.FileOrDefaults()
```

Comment:
Inline comment, `// we set fields in the raw file without writing env / flag overrides`